src/client/Client.cc

```diff
@@ -11516,13 +11516,23 @@ int64_t Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
 
   // copy into fresh buffer (since our write may be resub, async)
   bufferlist bl;
+  size_t total_appended = 0;
   if (buf) {
     if (size > 0)
       bl.append(buf, size);
   } else if (iov){
     for (int i = 0; i < iovcnt; i++) {
       if (iov[i].iov_len > 0) {
-        bl.append((const char *)iov[i].iov_base, iov[i].iov_len);
+        if (iov[i].iov_len >= static_cast<size_t>(size) 
+        || total_appended >= static_cast<size_t>(size)) {
```

Comment:
This could overshoot the clamp size. Shouldn't this check be:

```
if (iov[I].iov_len > ... || total_appended + iov[I].iov_len >= ...) {
    break;
}

Reply:
Or, even drop the first check.

Reply:
This part of patch not only makes sure that we don't over shoot but also makes sure we write up to to the size available to us (size - total_appended) where `size` can be at max `INT_MAX`.

If you look at the below code, only if the `total_appended` is not exceeding `size` (`INT_MAX`) is the time when we'd append `size - total_appended`. I can think of four cases:

1) if the incoming iov's size exceeds INT_MAX but the total size appended till now is within the limit:
- inner `if {}` only appends the size available i.e. `size - total_appended`

2) if the incoming iov is within limit but total size appended exceeds:
- inner `if {}` evaluates false and doesn't append

3) both 1 and 2 i.e. the incoming iov size and the total appended size both are > `INT_MAX`:
- inner `if {}` evaluates false

4) if none is true
- normal append takes place


Does this make sense?

Reply:
> If you look at the below code, only if the `total_appended` is not exceeding `size` (`INT_MAX`) is the time when we'd append `size - total_appended`

In that case, the check

> total_appended + iov[I].iov_len >= ...

is what's required right?

Reply:
you're right, the current code is unnecessarily complicated, what you suggested does the same but is much simpler to read

Reply:
@vshankar done, PTAL